atomic_reactor/plugins/post_import_image.py

```diff
@@ -77,30 +77,48 @@ def run(self):
                                   build_json_dir=self.build_json_dir,
                                   namespace=metadata.get('namespace', None))
         osbs = OSBS(osbs_conf, osbs_conf)
-
+        imagestream = None
         try:
-            osbs.get_image_stream(self.imagestream)
+            imagestream = osbs.get_image_stream(self.imagestream)
         except OsbsResponseException:
             if self.insecure_registry is not None:
                 kwargs['insecure_registry'] = self.insecure_registry
 
             self.log.info("Creating ImageStream %s for %s", self.imagestream,
                           self.docker_image_repo)
 
-            # Tags are imported automatically on creation
-            osbs.create_image_stream(self.imagestream, self.docker_image_repo,
-                                     **kwargs)
-        else:
-            self.log.info("Importing new tags for %s", self.imagestream)
-
-            attempts = 0
-            while not osbs.import_image(self.imagestream):
-                attempts += 1
+            imagestream = osbs.create_image_stream(self.imagestream,
+                                                   self.docker_image_repo,
+                                                   **kwargs)
+        self.log.info("Importing new tags for %s", self.imagestream)
 
-                if attempts >= self.import_attempts:
-                    msg = "Failed to import new tags for %s"
-                    raise RuntimeError(msg % self.imagestream)
-
-                self.log.info("no new tags, will retry after %d seconds (%d/%d)",
-                              self.retry_delay, attempts, self.import_attempts)
-                sleep(self.retry_delay)
+        primaries = []
+        try:
+            primaries = self.workflow.build_result.annotations['repositories']['primary']
```

Comment:
In the case where there are no `primaries`, do we want to bail out immediately? The `for-loop` below will be skipped since `primaries` is an empty list, but `osbs.import_image` will still be called.

If the use case for no `primaries` is only for scratch builds, then this is not a valid use case since the plugin is disabled for scratch builds.
I'm wondering if we should fail the plugin if there are no `primaries`, as that's not expected.
If not, then we should probably make sure that `osbs.import_image` is not called if there are not primaries.